packages/network-controller/tests/provider-api-tests/no-block-param.ts

```diff
@@ -0,0 +1,916 @@
+          comms.mockNextBlockTrackerRequest();
+          comms.mockRpcCall({
+            request,
+            error: errorMessage,
+          });
+          const promiseForResult = withNetworkClient(
+            { providerType, customRpcUrl },
+            async ({ makeRpcCall }) => makeRpcCall(request),
+          );
+
+          await expect(promiseForResult).rejects.toThrow(
+            buildFetchFailedErrorMessage(customRpcUrl, errorMessage),
+          );
+        },
+      );
+    });
+
+    it('retries the request to the RPC endpoint up to 5 times if a "failed to parse response body" error is thrown while making the request, returning the successful result if there is one on the 5th try', async () => {
+      await withMockedCommunications({ providerType }, async (comms) => {
+        const request = { method };
+
+        // The first time a block-cacheable request is made, the latest block
+        // number is retrieved through the block tracker first. It doesn't
+        // matter what this is — it's just used as a cache key.
+        comms.mockNextBlockTrackerRequest();
+        // Here we have the request fail for the first 4 tries, then succeed
+        // on the 5th try.
+        comms.mockRpcCall({
+          request,
+          error: 'failed to parse response body: some message',
+          times: 4,
+        });
+
+        comms.mockRpcCall({
+          request,
+          response: {
+            result: 'the result',
+            httpStatus: 200,
+          },
+        });
+
+        const result = await withNetworkClient(
+          { providerType },
+          async ({ makeRpcCall, clock }) => {
+            return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
+              makeRpcCall(request),
+              clock,
+            );
+          },
+        );
+
+        expect(result).toStrictEqual('the result');
+      });
+    });
+
+    it('returns an empty response if a "failed to parse response body" error is thrown while making the request to the RPC endpoint 5 times in a row', async () => {
+      await withMockedCommunications({ providerType }, async (comms) => {
+        const request = { method };
+        const errorMessage = 'failed to parse response body: some message';
+
+        // The first time a block-cacheable request is made, the latest block
+        // number is retrieved through the block tracker first. It doesn't
+        // matter what this is — it's just used as a cache key.
+        comms.mockNextBlockTrackerRequest();
+        comms.mockRpcCall({
+          request,
+          error: errorMessage,
+          times: 5,
+        });
+        const result = await withNetworkClient(
+          { providerType },
+          async ({ makeRpcCall, clock }) => {
+            return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
+              makeRpcCall(request),
+              clock,
+            );
+          },
+        );
+
+        expect(result).toBeUndefined();
```

Comment:
Again this behavior is different from the extension, which produces a "response is empty" error.